### DISCUSSION TITLE
Install Question

### DISCUSSION THREAD

**Discussion by skrilla0** [toverainc/willow#130] [General]
First of all, I'm excited to get this going, and have been following along in the community. 

I'm at the point of the README.MD where I am finally ready to flash the build. I'm a bit of a newbie here, but after identifying where the USB is located on my Linux box, running the export command, I'm having trouble setting the PORT environment variable when trying to erase the factory flash. Receiving the error: `You need to define the PORT environment variable to do serial stuff - exiting`

Help on what I'm missing? Thanks

**Comment by nikito** [comment]
I think the issue is that when you do an export as your regular user, then use sudo to run the script, it does not preserve your user's exports. There's two ways to get around that from what I remember.
1. Set the variable inline with your command. So for example: sudo PORT=/dev/ttyACM0 ./utils.sh erase-flash
2. Do the export, then use the -E option with Sudo (-E tells sudo to preserve all of the user's environment variables when running). So you could do: export PORT=/dev/ttyACM0; sudo -E ./utils-sh erase-flash

Of these options I believe the first one is best (using -E for sudo exposes ALL user variables, and may expose unexpected/undersired varibles, whereas the first option exposes only the specific one(s) you are setting in the command).
I haven't had a chance to try either of these myself yet as the S3 boxes are impossible to come by at the moment 🤣

**Comment by skrilla0** [comment]
Thank you @nikito this was exactly it. Really appreciate your help.

**Reply by nikito** [reply]
Awesome, glad I could help!
